ruby/red-arrow/benchmark/raw_records/dict.yml

```diff
@@ -0,0 +1,68 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+prelude: |-
+  require "arrow"
+  require "faker"
+
+  state = ENV.fetch("FAKER_RANDOM_SEED", 17).to_i
+  Faker::Config.random = Random.new(state)
+
+  n_rows = 1000
+  n_columns = 10
+  dictionary = Arrow::StringArray.new(
+    100.times.map { Faker::Book.genre }.uniq
+  )
+  type = Arrow::DictionaryDataType.new(:int8, dictionary, true)
+
+  fields = n_columns.times.map {|i| ["column_#{i}".to_sym, type] }.to_h
+  schema = Arrow::Schema.new(**fields)
+  arrays = n_columns.times.map do
+    Arrow::DictionaryArray.new(
+      type,
+      Arrow::Int8Array.new(
+        n_rows.times.map {
+          Faker::Number.within(0 ... dictionary.length)
+        }
+      )
+    )
+  end
+  record_batch = Arrow::RecordBatch.new(schema, n_rows, arrays)
```

Comment:
Could you add a TODO comment that we need to rewrite this code when ARROW-4671 and ARROW-4776 are resolved?